app/src/main/java/com/github/polybooks/database/InterestDatabase.kt

```diff
@@ -41,23 +41,21 @@ interface InterestDatabase {
     /**
      * List all the interests in the database.
      * */
-    fun listAllInterests() : CompletableFuture<Triple<List<Field>, List<Semester>, List<Course>>>
-        = listAllFields().thenCombine(listAllSemesters()) { fields, semester -> Pair(fields, semester) }
+    fun listAllInterests() : CompletableFuture<Triple<List<Topic>, List<Semester>, List<Course>>>
+        = listAllTopics().thenCombine(listAllSemesters()) { topics, semester -> Pair(topics, semester) }
         .thenCombine(listAllCourses()){pair, courses -> Triple(pair.first, pair.second, courses)}
     /**
-     * Get the interests of the specified user
-     * TODO: Might need to add an authentication token to restrict authenticated users to only modify their interests.
+     * Get the interests of the current user.
+     * If the user is not auth, it will use exclusively the local storage. (TODO)
```

Comment:
Don't forget to remove the TODO once it's done :)

Reply:
Cf my previous answer: it will not be done. But that's a conversation we could have all together:
Do we keep TODOs for the final deadline that show concepts and intention of features, and that might potentially be useful if someone decide to continue the project outside of the course?
Or do we clean up all TODOs?